### ISSUE TITLE
timeline - select (and move) multiple clips

### ISSUE THREAD

**Issue by faridosc** [jliljebl/flowblade#146]
the possibility to click and drag with the mouse to select various clips and move them around in the timeline. maybe implement post gtk3 port?

**Comment by jliljebl** [comment]
This would be absolutely necessary if selecting clips from multiple tracks was needed in Flowblade, but with current design having this would mostly be a matter of user preference. Some points:

* Selecting a clip range with "CTRL + Mouse Left" requires less tensioned mouse movement and is thus less straining on mouse hand
* currently pressing on clip and dragging initiates an immediate moving of the clip which is convenient
* an early motto for this design was "no vertical scrolling, no box selection" and I'm quite happy with this design approach

I'm open to suggestions on how selecting clip range with mouse should exactly be done, and i leave this open for further discussion, but I'm not planning on doing this in the near future

**Comment by jliljebl** [comment]
This going wontfix, functionality is available using  "CTRL + Mouse Left" , multitrack selection is not needed, and mouse drag box selection for single track is not in the spirit of the design.

**Comment by schaaff** [comment]
It is needed!
How do I move two synchronized clips having "Image in Image" transition in parallel without loosing the sync!!
How do I move the clip with its corresponding transition.

In my opinion, this is more then necessary, but maybe the usability concept is thought for more intellectual.

**Comment by jliljebl** [comment]
The proposed workflow is shown here: https://vimeo.com/161526014

Some points:

1) One Flowblade is fundamentally an INSERT paradigm editing program meaning that cutting, trimming and moving clips around is very fast and convenient, but this creates an unavoidable issue:  most edits cause tracks to lose sync with each other, and this is a much bigger issue then anything related to multitrack box selection. E.g a typical Avid editing view presents multiple GUI elements warning that clips have changed sync between each other after original edits placed them in some position in relation to each other.

2) Combining insert editing with clip grouping and/or box selection has been done in Final Cut X but the result was VERY complex from both usability and programming POV, and has not generally been well received.

3) Flowblade has chosen to present user with tools to explicitly create and regain sync relations between clips and compositors.

4) Multitrack box selections would not help at all with the more important inherent problem of losing sync between tracks present in both Avid and Flowblade.

6) Use the MOve tool to resync single clips and Spacer tool with CTRL to all the later clips in a single adit

**Comment by gamnamno** [comment]
The "sync parent" option is cool, but how about adding an option to make it automatic ?

**Comment by gamnamno** [comment]
I don't understand your 6) part though. Which mouse button do you use and what's "point", and do you mean "single edit" ?

**Comment by jliljebl** [comment]
@gamnamno Automatic syncing would break user expectations on how Flowblade works - with simple and explicit edit actions.

It will result in Final Cut X type difficultly controllable work flow.

On point 6) I simply suggest using CTRL + "Move" tool to move all clips in a track without having to select them all.

**Comment by gamnamno** [comment]
Ok, thanks.
Now one more question : why can't we CTRL+LMB to select clips on different tracks ?

**Comment by gamnamno** [comment]
Also you said :
> Automatic syncing would break user expectations
but I proposed it to be an option. That may not be the default, but there could be an on/off switch for it somewhere in the toolbar or in the prefs.

**Comment by jliljebl** [comment]
Autosyncing would require modifying code in 50+ places in my estimation. Its basically a full rewrite of all the edit code and I'm not going in that direction. The end result would be very Final Cut X like and I'm not a fan.

>  but why not allow to use this shortcut on different tracks ?

Pitivi has this and it's horrible.

There simply will not be multitrack selections in Flowblade, ever.

The use case people wish to achieve with multitrack selections is to be able to move sequences of multitrack compositions and/or audio split edits intact in a single edit.

The problem is that any attempt to do this with box selections will, again, end up in Final Cut X land where clips jump around or get overwritten in very complex ways.

The only way to do this in my view would be to create some special **Sequence Move** tool, that would work something like the following:
* User selects a range of clips in track *V1* (again using CTRL + Left Mouse, not box drag), and **only** on track V1
* This selection would define a group of clips consisting of all clips and compositors fully contained in the timeline range defined by clips in track V1.
* this group could then be inserted in any edit point on track V1 and only on track V1.  Additional restrictoin would be that no clips may go ever the frame of insert frame on track V1

 I added Issue #270 so this be attempted sometime, but not in a very close future.

**Comment by gamnamno** [comment]
Cool, that's a great idea !
